### PR TITLE
[Fix] fix a bug about validation of ava dataset

### PR DESCRIPTION
**Pull request by irvingzhang0512**
+ Description: When training ava dataset with default configs, no validaton results stored.
+ Multi-GPU
  + script: `tools/dist_train.sh configs/detection/ava/slowonly_kinetics_pretrained_r50_4x16x1_20e_ava_rgb.py 8 --validate`
  + details: successfully run eval after every epoch, but `best.json` is not created in work dir.
+ single-gpu:
  + script: `python tools/train.py configs/detection/ava/slowonly_kinetics_pretrained_r50_4x16x1_20e_ava_rgb.py --validate`
  + details: sucessfully run ava eval, but `TypeError` occured.

```
/ssd01/zhangyiyang/mmaction2_github/mmaction/core/evaluation/eval_hooks.py:176: UserWarning: The key indicator for evaluation is not included in evaluation result, please specify it in config file
  warnings.warn('The key indicator for evaluation is not '
Traceback (most recent call last):
  File "tools/train.py", line 178, in <module>
    main()
  File "tools/train.py", line 174, in main
    meta=meta)
  File "/ssd01/zhangyiyang/mmaction2_github/mmaction/apis/train.py", line 156, in train_model
    runner.run(data_loaders, cfg.workflow, cfg.total_epochs, **runner_kwargs)
  File "/home/ubuntu/anaconda3/envs/zyy_pytorch1.6/lib/python3.7/site-packages/mmcv/runner/epoch_based_runner.py", line 125, in run
    epoch_runner(data_loaders[i], **kwargs)
  File "/home/ubuntu/anaconda3/envs/zyy_pytorch1.6/lib/python3.7/site-packages/mmcv/runner/epoch_based_runner.py", line 54, in train
    self.call_hook('after_train_epoch')
  File "/home/ubuntu/anaconda3/envs/zyy_pytorch1.6/lib/python3.7/site-packages/mmcv/runner/base_runner.py", line 307, in call_hook
    getattr(hook, fn_name)(self)
  File "/ssd01/zhangyiyang/mmaction2_github/mmaction/core/evaluation/eval_hooks.py", line 153, in after_train_epoch
    if (self.save_best and self.compare_func(key_score, self.best_score)):
  File "/ssd01/zhangyiyang/mmaction2_github/mmaction/core/evaluation/eval_hooks.py", line 46, in <lambda>
    rule_map = {'greater': lambda x, y: x > y, 'less': lambda x, y: x < y}
TypeError: '>' not supported between instances of 'NoneType' and 'float'
```

+ Locating BUG: `key_indicator` for eval hook is not set. Default `key_indicator` is `top1_acc`, which is useless for ava dataset.

+ Modification:
    + set default `key_indicator='mAP@0.5IOU'` for ava dataset configs.
    + rename ava metrics `PascalBoxes_Precision/mAP@0.5IOU` to `mAP@0.5IOU`
    + modify eval hook, add docs & keys for `greater_keys`

### CODE REVIEW

**Comment by innerlee** [comment]
Another evidence that our tool scripts need tests

**Comment by dreamerlin** [comment]
LGTM

**Review by dreamerlin — APPROVED** [review]

File: configs/detection/ava/slowfast_context_kinetics_pretrained_r50_4x16x1_20e_ava_rgb.py [hunk 1]
```diff
@@ -159,7 +159,12 @@
 total_epochs = 20
 checkpoint_config = dict(interval=1)
 workflow = [('train', 1)]
-evaluation = dict(interval=1)
+# Three avaiable key indicators for ava dataset
+# 'Recall@0.5@100' 'AR@100' 'PascalBoxes_Precision/mAP@0.5IOU'
```
Comment (innerlee) [inline configs/detection/ava/slowfast_context_kinetics_pretrained_r50_4x16x1_20e_ava_rgb.py:163]: Where does the name `PascalBoxes_Precision` come from?

Comment (irvingzhang0512) [reply configs/detection/ava/slowfast_context_kinetics_pretrained_r50_4x16x1_20e_ava_rgb.py:163]: I think this comes from official ava evalution codes.

https://github.com/open-mmlab/mmaction2/blob/023777cfd26bb175f85d78c455f6869673e0aa09/mmaction/core/evaluation/ava_evaluation/object_detection_evaluation.py#L307-L308

And `self._metric_prefix` is defined 

https://github.com/open-mmlab/mmaction2/blob/023777cfd26bb175f85d78c455f6869673e0aa09/mmaction/core/evaluation/ava_evaluation/object_detection_evaluation.py#L151

https://github.com/open-mmlab/mmaction2/blob/023777cfd26bb175f85d78c455f6869673e0aa09/mmaction/core/evaluation/ava_evaluation/object_detection_evaluation.py#L356

**Review by kennymckormick — COMMENTED** [review]
Two suggestions:
1. What about we use a shorter name like `mAP@0.5IOU` instead of the current long name `PascalBoxes_Precision/mAP@0.5IOU`? For example, we can change the name in AVADataset.evaluate
2. What about we add the new key_indicator to the greater_keys ([here](https://github.com/kennymckormick/mmaction2/blob/611ae31515612c33c73141e217e3a23b0f1a9c2f/mmaction/core/evaluation/eval_hooks.py#L48)), so that we don't need to wrtie `rule='greater'` for each config

**Comment by irvingzhang0512** [comment]
@dreamerlin #395 need to be updated (spatio-temporal action detection related docs and `greater_keys`)

File: mmaction/core/evaluation/ava_evaluation/object_detection_evaluation.py [hunk 2]
```diff
@@ -304,7 +304,7 @@ def evaluate(self):
             mean_corloc,
         ) = self._evaluation.evaluate()
 
-        metric = 'Precision/mAP@{}IOU'.format(self._matching_iou_threshold)
+        metric = 'mAP@{}IOU'.format(self._matching_iou_threshold)
```
Comment (innerlee) [inline mmaction/core/evaluation/ava_evaluation/object_detection_evaluation.py:307]: use f-string

**Review by kennymckormick — APPROVED** [review]

**Pull request merged**